controllers/ajax.py

```diff
@@ -176,7 +176,8 @@ def getprog():
         result = db(query)
         res['acid'] = acid
         if not result.isempty():
-            r = result.select(orderby=~codetbl.timestamp).first().code
+            # get the last code they saved; id order gets that for us
+            r = result.select(orderby=codetbl.id).last().code
```

Comment:
So I guess ordering by id is just as reliable as ordering by timestamp, but avoids the null problem.